### ISSUE TITLE
Retry app-store-connect fetch-signing-files on error?

### ISSUE THREAD

**Issue by kolotum** [codemagic-ci-cd/cli-tools#171]
We have the following step in our codemagic.yaml file:

```
- name: Fetch signing files
  script: app-store-connect fetch-signing-files "$BUNDLE_ID" --type IOS_APP_STORE --create
```

Sometimes the build fails with the following error message:

> GET https://api.appstoreconnect.apple.com/v1/bundleIds?limit=100&sort=name&filter%5Bidentifier%5D=bundle.id.redacted&filter%5Bplatform%5D=IOS returned 401: Authentication credentials are missing or invalid. - Provide a properly configured and signed bearer token, and make sure that it has not expired. Learn more about Generating Tokens for API Requests https://developer.apple.com/go/?id=api-generating-tokens

When we start the same build it usually works again. So it seems to be a temporary issue at api.appstoreconnect.apple.com.

Is there an easy way to retry the fetch-signing-files on error?

An additional parameter could be helpful:
`app-store-connect fetch-signing-files "$BUNDLE_ID" --type IOS_APP_STORE --create --retries 3`

**Comment by bohdan1krokhmaliuk** [comment]
Have same issue.
It doesn't relates to fetch-signing-files only. It is general problem,
Sometimes it works fine and sometimes it looks like apple servers have some issues.

Looks like there was similar [issue](https://github.com/fastlane/fastlane/issues/19072) with a fastlane

**Comment by priitlatt** [comment]
Will be resolved by #184.
